cpp/src/db/db_execution.cu

```diff
@@ -0,0 +1,825 @@
+    uint32_t rootChildren = cypher_astnode_nchildren(root);
+    CUGRAPH_EXPECTS(rootChildren == 1, "Parse result root has more than one child");
+
+    // Get the first child of the root node, expected to be a 'query'
+    const cypher_astnode_t* queryNode = cypher_astnode_get_child(root, 0);
+    std::string queryType             = getTypeString(queryNode);
+    CUGRAPH_EXPECTS(queryType == "query", "Expected query node, got something else");
+
+    // For each child of the 'query' node, construct an execution_node for it
+    uint32_t queryChildren = cypher_astnode_nchildren(queryNode);
+    for (uint32_t j = 0; j < queryChildren; j++) {
+      const cypher_astnode_t* child = cypher_astnode_get_child(queryNode, j);
+      std::string childType         = getTypeString(child);
+      if (childType == "LOAD CSV") {
+        execution_node<idx_t>* childPtr = new load_csv_node<idx_t>(child);
+        plan_nodes.push_back(childPtr);
+      } else if (childType == "MATCH") {
+        execution_node<idx_t>* childPtr = new match_node<idx_t>(child, ctx);
+        plan_nodes.push_back(childPtr);
+      } else if (childType == "CREATE") {
+        execution_node<idx_t>* childPtr = new create_node<idx_t>(child, ctx);
+        plan_nodes.push_back(childPtr);
+      } else {
+        CUGRAPH_FAIL("Unsupported query clause given");
+      }
+    }
+  }
+}
+
+template <typename idx_t>
+query_plan<idx_t>::query_plan(query_plan&& other)
+{
+  ctx        = std::move(other.ctx);
+  plan_nodes = std::move(other.plan_nodes);
+  other.plan_nodes.clear();
+}
+
+template <typename idx_t>
+query_plan<idx_t>::~query_plan()
+{
+  for (size_t i = 0; i < plan_nodes.size(); i++) delete plan_nodes[i];
+  plan_nodes.clear();
+}
+
+template <typename idx_t>
+query_plan<idx_t>& query_plan<idx_t>::operator=(query_plan<idx_t>&& other)
+{
+  if (this != &other) {
+    ctx = std::move(other.ctx);
+    for (size_t i = 0; i < plan_nodes.size(); i++) delete plan_nodes[i];
+    plan_nodes = std::move(other.plan_nodes);
+    other.plan_nodes.clear();
+  }
+  return *this;
+}
+
+template <typename idx_t>
+std::string query_plan<idx_t>::execute()
+{
+  for (size_t i = 0; i < plan_nodes.size(); i++) {
+    execution_type type = plan_nodes[i]->type();
+    switch (type) {
+      case execution_type::LoadCsv: {
+        load_csv_node<idx_t>* csv_node = reinterpret_cast<load_csv_node<idx_t>*>(plan_nodes[i]);
+        csv_node->execute(ctx);
+        break;
+      }
+      case execution_type::Match: {
+        break;
+      }
+      case execution_type::Merge: {
+        break;
+      }
+      case execution_type::Create: {
+        std::cout << "Calling create_node<idx_t>::execute()\n";
+        create_node<idx_t>* c_node = reinterpret_cast<create_node<idx_t>*>(plan_nodes[i]);
+        c_node->execute(ctx);
+        break;
+      }
+      case execution_type::Return: {
+        break;
+      }
+    }
+  }
+  std::stringstream ss;
+  return ss.str();
```

Comment:
What's the purpose stringstream here?

Reply:
It's there to be used when constructing the output of the method. It wasn't being used because the parts of the query plan that write out results are not implemented yet.

Reply:
It's a placeholder for code which hasn't been implemented yet.